File: src/main/quarter/net/coderextreme/data/AllenBox.cpp

#include <Inventor/nodes/SoCone.h>
#include <Inventor/nodes/SoBaseColor.h>
#include <Inventor/nodes/SoSeparator.h>
#include <Inventor/nodes/SoShape.h>
#include <Inventor/nodes/SoTransform.h>
#include <Inventor/nodes/SoSphere.h>
#include <Inventor/nodes/SoMaterial.h>
#include <Inventor/misc/SoProtoInstance.h>
#include <Inventor/VRMLnodes/SoVRMLAppearance.h>
#include <Quarter/Quarter.h>
#include <Quarter/QuarterWidget.h>
#include <QApplication>
#include <QMainWindow>

using namespace SIM::Coin3D::Quarter;
int main(int argc, char ** argv) 
{

  QApplication app(argc, argv);
  Quarter::init();

  SoSeparator * root = new SoSeparator;
  root->ref();

  SoBaseColor * col = new SoBaseColor;
  col->rgb = SbColor(1, 1, 0);
  root->addChild(col);

  root->addChild(new SoCone);

  QMainWindow * mainwin = new QMainWindow();

  QuarterWidget * viewer = new QuarterWidget(mainwin);
  viewer->setNavigationModeFile();
  mainwin->setCentralWidget(viewer);
  viewer->setSceneGraph(root);

  mainwin->show();
  app.exec();
  root->unref();
  delete viewer;

  Quarter::clean();
SoSceneManager* SoSceneManager0 = new SoSceneManager();
SoSceneManager0->setProfile(QString("Immersive"));
SoSceneManager0->setVersion(QString("4.0"));
Sohead* Sohead1 = new Sohead();
Someta* Someta2 = new Someta();
Someta2->setName(QString("title"));
Someta2->setContent(QString("AllenBox.x3d"));
Sohead1->addMeta(*Someta2);

Someta* Someta3 = new Someta();
Someta3->setName(QString("creator"));
Someta3->setContent(QString("Allen Box"));
Sohead1->addMeta(*Someta3);

Someta* Someta4 = new Someta();
Someta4->setName(QString("created"));
Someta4->setContent(QString("8 July 2025"));
Sohead1->addMeta(*Someta4);

Someta* Someta5 = new Someta();
Someta5->setName(QString("license"));
Someta5->setContent(QString("../license.html"));
Sohead1->addMeta(*Someta5);

SoSceneManager0->setHead(*Sohead1);

SoNode* SoNode6 = new SoNode();
SoWorldInfo* SoWorldInfo7 = new SoWorldInfo();
SoWorldInfo7->setTitle(QString("AllenBox.x3d"));
SoNode6->addChild(*SoWorldInfo7);

SoNavigationInfo* SoNavigationInfo8 = new SoNavigationInfo();
SoNavigationInfo8->setAvatarSize(new float[]{0.15,1.53,0.75}, 3);
SoNavigationInfo8->setSpeed(0.5);
SoNavigationInfo8->setType(new QString[]{QString("EXAMINE")}, 1);
SoNode6->addChild(*SoNavigationInfo8);

SoTransform* SoTransform9 = new SoTransform();
SoTransform9->setDEF(QString("Floor"));
SoTransform9->setScale(new float[]{1.0,0.0125,1.0});
SoTransform9->setTranslation(new float[]{0.0,-0.0125,0.0});
SoShape* SoShape10 = new SoShape();
SoBox* SoBox11 = new SoBox();
SoShape10->setGeometry(*SoBox11);

SoVRMLAppearance* SoVRMLAppearance12 = new SoVRMLAppearance();
SoMaterial* SoMaterial13 = new SoMaterial();
SoVRMLAppearance12->addChild(*SoMaterial13);

SoShape10->addChild(*SoVRMLAppearance12);

SoTransform9->addChild(*SoShape10);

SoNode6->addChild(*SoTransform9);

SoSceneManager0->setSceneGraph(*SoNode6);

return 0;
}

File: src/main/graaljs/net/coderextreme/data/AllenBox.js
load('X3Dautoclass.js');
var ConfigurationProperties = Packages.org.web3d.x3d.jsail.ConfigurationProperties;
ConfigurationProperties.setXsltEngine(ConfigurationProperties.XSLT_ENGINE_NATIVE_JAVA);
ConfigurationProperties.setDeleteIntermediateFiles(false);
ConfigurationProperties.setStripTrailingZeroes(true);
ConfigurationProperties.setStripDefaultAttributes(true);
function doubleToFloat(d) {
    if (Float32Array)
	return new Float32Array(d);
}
      var X3D0 =  new X3D().setProfile("Immersive").setVersion("4.0")
      .setHead(new head()
        .addMeta(new meta().setName("title").setContent("AllenBox.x3d"))
        .addMeta(new meta().setName("creator").setContent("Allen Box"))
        .addMeta(new meta().setName("created").setContent("8 July 2025"))
        .addMeta(new meta().setName("license").setContent("../license.html")))
      .setScene(new Scene()
        .addChild(new WorldInfo().setTitle("AllenBox.x3d"))
        .addChild(new NavigationInfo().setType(Java.to(["EXAMINE"], Java.type("java.lang.String[]"))).setAvatarSize(Java.to(doubleToFloat([0.15,1.53,0.75]), Java.type("float[]"))).setSpeed(0.5))
        .addChild(new Transform().setDEF("Floor").setScale(Java.to(doubleToFloat([1,0.0125,1]), Java.type("float[]"))).setTranslation(Java.to(doubleToFloat([0,-0.0125,0]), Java.type("float[]")))
          .addChild(new Shape()
            .setGeometry(new Box())
            .setAppearance(new Appearance()
              .setMaterial(new Material())))))      ;
    X3D0.toFileX3D("../data/AllenBox.new.graal.x3d");
    X3D0.toFileJSON("../data/AllenBox.new.graal.json");
